mongo/client.go

```diff
@@ -437,7 +437,7 @@ func (c *Client) configureAutoEncryption(clientOpts *options.ClientOptions) erro
 	}
 
 	// If the crypt_shared library was loaded successfully, signal to the mongocryptd client creator
-	// that it can bypass spawning mongocryptd.
+	// that it can bypass spawning or connecting to mongocryptd.
 	cryptSharedLibAvailable := mc.CryptSharedLibVersionString() != ""
 	mongocryptdFLE, err := newMongocryptdClient(cryptSharedLibAvailable, clientOpts.AutoEncryptionOptions)
```

Comment:
Is there anything in the `mongocryptdClient` that we need when the crypt_shared library is loaded? Can we skip creating the client completely?

E.g.
```go
if mc.CryptSharedLibVersionString() == "" {
	mongocryptdFLE, err := newMongocryptdClient(clientOpts.AutoEncryptionOptions)
	if err != nil {
		return err
	}
	c.mongocryptdFLE = mongocryptdFLE
}
```

Reply:
I think no. Skipping creating the client seems preferable. Thank you for the suggested code. Updated.